facilities/models.py

```diff
@@ -0,0 +1,83 @@
+import random
+from django.db import models
+from common.models import AbstractBase, SubCounty
+
+
+class Owner(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def generate_code(self):
+        return "{}{}".format("OWNER", str(self.id))
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = self.generate_code()
+        super(Owner, self).save(*args, **kwargs)
+
+
+class Service(AbstractBase):
+    name = models.CharField(max_length=255)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def __unicode__(self):
+        return self.name
+
+    def generate_code(self):
+        return "{}{}".format(self.name, str(self.id))
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = self.generate_code()
+        super(Service, self).save(*args, **kwargs)
+
+
+class GIS(AbstractBase):
+    latitude = models.CharField(max_length=255)
+    longitude = models.CharField(max_length=255)
+    is_classified = models.BooleanField(default=False)
+
+
+class FacilityStatus(AbstractBase):
+    name = models.CharField(max_length=100)
+
+
+class FacilityType(AbstractBase):
+    name = models.CharField(max_length=100)
+
+
+class Facility(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    code = models.CharField(max_length=100, unique=True)
```

Comment:
( see the comment on the `generate_code` function first )

It will be easier - even on testing - if you make `generate_code` a module level no-args callable that is then supplied as the `default` for the `code` field.

In order to distinguish it from other code generation callables, give it a descriptive name e.g `generate_sequential_facility_code`. I suspect that you will have other code generating functions, and that opportunities for refactoring / code sharing will be seen without too much effort.